src/openrct2/actions/RideDemolishAction.cpp

```diff
@@ -130,14 +130,15 @@ GameActions::Result::Ptr RideDemolishAction::DemolishRide(Ride* ride) const
 
     sub_6CB945(ride);
     ride_clear_leftover_entrances(ride);
-    News::DisableNewsItems(News::ItemType::Ride, _rideIndex);
+
+    News::DisableNewsItems(News::ItemType::Ride, static_cast<uint32_t>(_rideIndex));
 
     UnlinkAllBannersForRide(_rideIndex);
 
     for (auto peep : EntityList<Guest>())
     {
-        uint8_t ride_id_bit = _rideIndex % 8;
-        uint8_t ride_id_offset = _rideIndex / 8;
+        uint8_t ride_id_bit = static_cast<uint32_t>(_rideIndex) % 8;
```

Comment:
This is obviously broken for >255 rides but is fixed in my pr for the ride use memory. I'm only putting this note here for reference in case anyone else wonders.